### PR TITLE
EVG-6380 add method to find role with given permissions

### PR DESCRIPTION
**Pull request by john-m-liu**
- Assume that scopes always have all their relevant resources listed within them
- Add a method to find a role with given permissions. This isn't used yet, but will be used by MANA to find an existing role that matches what permissions the user requests

### CODE REVIEW

File: rolemanager/role_manager.go [hunk 1]
```diff
@@ -181,6 +151,35 @@ func (m *mongoBackedRoleManager) FilterScopesByResourceType(ScopeIDs []string, r
...
+		"$and": []bson.M{
+			{"resources": bson.M{
+				"$all": resources,
+			}},
+			{"resources": bson.M{
+				"$size": len(resources),
+			}},
+		},
+	}
+	cursor, err := coll.Find(ctx, query)
```
Comment (julianedwards) [inline rolemanager/role_manager.go]: Couldn't we just do a `FindOne` here?

File: auth_interfaces.go [hunk 2]
```diff
@@ -102,4 +105,10 @@ type RoleManager interface {
 	// RegisterPermissions adds a list of strings to the role manager as valid permission keys. Returns an
 	// error if the same permission is registered more than once
 	RegisterPermissions([]string) error
+
+	// FindRolesWithPermissions returns a role that exactly matches the given resources and permissions
+	FindRoleWithPermissions(string, []string, Permissions) (*Role, error)
```
Comment (julianedwards) [inline auth_interfaces.go]: maybe `FindRoleWithResourcesAndPermissions` ?

File: rolemanager/role_manager.go (hunk 1 above)
Comment (john-m-liu) [reply rolemanager/role_manager.go]: oops fixed

File: auth_interfaces.go (hunk 2 above)
Comment (john-m-liu) [reply auth_interfaces.go]: It seems too wordy to me. I think it's implied that you need to pass specific resources
